spark-atlas-connector/src/main/scala/com/hortonworks/spark/atlas/types/external.scala

```diff
@@ -144,6 +147,44 @@ object external {
     Seq(dbEntity)
   }
 
+  // ================ RDBMS based entities ======================
+  val RDBMS_TABLE = "rdbms_table"
+
+  /**
+   * Converts JDBC RDBMS properties into Atlas entity
+   *
+   * @param url
+   * @param tableName
+   * @return
+   */
+  def rdbmsTableToEntity(url: String, tableName: String) : Seq[AtlasEntity] = {
+    val jdbcEntity = new AtlasEntity(RDBMS_TABLE)
+
+    val databaseName = getRdbmsDatabaseName(url)
+    jdbcEntity.setAttribute(QUALIFIED_NAME_KEY, getRdbmsQualifiedName(databaseName, tableName))
+    jdbcEntity.setAttribute(ATTRIBUTE_NAME_KEY, tableName)
+    Seq(jdbcEntity)
+  }
+
+  /**
+   * Constructs the the full qualified name of the databse
+   *
+   * @param databaseName
+   * @param tableName
+   * @return
+   */
+  private def getRdbmsQualifiedName(databaseName: String, tableName: String): String =
+    s"${databaseName.toLowerCase}.${tableName.toLowerCase}"
+
+  /**
+   * Gets the database in use by the RDBMS JDBC
+   *
+   * @param url The URL used by a JDBC connector
+   * @return
+   */
+  private def getRdbmsDatabaseName(url: String): String =
+    url.substring(url.lastIndexOf("/") + 1)
```

Comment:
This is easy to be fragile. JDBC URL formats are different between DBMS, as well as even it has database as last part (one of example would be mysql) it allows adding properties which will be `?k1=v1&k2=v2&...` which will be included to database name here.

I don't mean we need to support all the DBMS JDBC URLs, but I would feel safer if we support at least MySQL (MariaDB)/PostgreSQL/Oracle/MS SQL Server. (I think we need to parse URL a bit and check which RDBMS the URL represents)

It's OK if we can't get database name safely from these list: then I think we end up with simply not supporting it (might be better to log some message as DEBUG or INFO level). Unlike others, we may need to have `Option[Seq[AtlasEntity]]` as return type of `rdbmsTableToEntity` because of this.

Reply:
Thanks for pointing this out. I had a bit of look, [and if I'm reading this correctly](https://github.com/apache/spark/blob/670bc55f8d357a5cd894e290cc2834e952a7cfe0/sql/core/src/main/scala/org/apache/spark/sql/jdbc/JdbcDialects.scala#L223), Spark currently supports 
- DB2Dialect
- DerbyDialect
- MsSqlDialect
- MySQLDialect
- OracleDialect
- PostgresDialect
- TeradataDialect

I'll write up a function to parse this, that will support the following DBMS.